packages/sirius-web/frontend/sirius-web-application/src/modals/duplicate-object/DuplicateObjectModal.tsx

```diff
@@ -0,0 +1,270 @@
+              }
+              onTextClear={() => setState((prevState) => ({ ...prevState, filterBarText: '' }))}
+              text={state.filterBarText}
+            />
+            <span className={classes.title}>Select the container</span>
+            <div className={classes.borderStyle} data-testid="duplicate-object-tree">
+              {tree !== null ? (
+                <TreeView
+                  editingContextId={editingContextId}
+                  readOnly={true}
+                  treeId={treeId}
+                  tree={tree}
+                  enableMultiSelection={false}
+                  synchronizedWithSelection={true}
+                  textToFilter={state.filterBarText}
+                  textToHighlight={state.filterBarText}
+                  treeItemActionRender={(props) => <ExpandAllTreeItemAction {...props} />}
+                  onExpandedElementChange={onExpandedElementChange}
+                  expanded={state.expanded}
+                  maxDepth={state.maxDepth}
+                />
+              ) : null}
+            </div>
+            <span className={classes.title}>Select the containment reference name</span>
+            <Select
+              value={state.containmentFeatureName}
+              onChange={onFeatureNameChange}
+              fullWidth
+              data-testid="containment-feature-name">
+              {state.containmentFeatureNames.map((featureName) => (
+                <MenuItem value={featureName} key={featureName}>
+                  {featureName}
+                </MenuItem>
+              ))}
+            </Select>
+            <FormControlLabel
+              control={
+                <Checkbox
+                  checked={state.duplicateContent}
+                  onChange={onIsDuplicateContentChange}
+                  name="duplicateContent"
+                  data-testid="duplicateContent"
+                />
+              }
+              label="Duplicate the content"
+            />
+            <FormControlLabel
+              control={
+                <Checkbox
+                  checked={state.copyOutgoingReferences}
+                  onChange={onIsCopyOutgoingReferencesContentChange}
+                  name="copyOutgoingReferences"
+                  data-testid="copyOutgoingReferences"
+                />
+              }
+              label="Copy outgoing references"
+            />
+            <FormControlLabel
+              control={
+                <Checkbox
+                  checked={state.updateIncomingReferences}
+                  onChange={onIsUpdateIncomingReferencesChange}
+                  name="updateIncomingReferences"
+                  data-testid="updateIncomingReferences"
+                />
+              }
+              label="Update incoming references"
+            />
+          </div>
+        </DialogContent>
+        <DialogActions>
+          <Button
+            variant="contained"
+            disabled={!state.containmentFeatureName}
```

Comment:
Use the loading state that should be provided by the useDuplicateObject modal to disable the dialog while the operation is being performed.